### PR TITLE
Add USAFacts to get NYC boroughs cases+deaths. Aggregate to new "place" region

### PR DESCRIPTION
**Pull request by TomGoBravo**
## This PR

* adds USAFacts as backup data source of cases and deaths
* Adds new AggregateLevel PLACE
* Changes NYC aggregation from boroughs to produce data with FIPS 3651000 and location id `iso1:us#iso2:us-ny#fips:3651000`
* Drops NY City data from NYTimes data source so NY County data from USAFacts appears in combined datasets
* Updates combined data to add the county boroughs and place NYC with `./run.py data update`
* Updates weights to add the county boroughs and place NYC with `./run.py data update-case-based-icu-utilization-weights`
* In calls to `get_icu_timeseries_from_regional_input` remove default args that were never used, making the source code closer to what is run

## Tested

Ran `./run.py data update` then `git difftool -t vimdiff master -- data/multiregion-wide-dates.csv` which shows 
* case/deaths/new_cases appearing for some tiny counties (iso1:us#iso2:us-ak#fips:02105, iso1:us#iso2:us-ak#fips:02282, iso1:us#iso2:us-hi#fips:15005
* The boroughs appear with a few stats:
```
iso1:us#iso2:us-ny#fips:36005,cases,USAFacts,,,,,,,,,,,,
iso1:us#iso2:us-ny#fips:36005,deaths,USAFacts,,,,,,,,,,,
iso1:us#iso2:us-ny#fips:36005,negative_tests,Valorum,,,,
iso1:us#iso2:us-ny#fips:36005,new_cases,,,,,,,,,,,,,,,,,
iso1:us#iso2:us-ny#fips:36005,positive_tests,Valorum,,,,
iso1:us#iso2:us-ny#fips:36005,test_positivity,CMSTesting
```
* FIPS 36061  is replaced with NY County data, looking like the above.
* Changes in `data/multiregion-static.csv` look similar
* `iso1:us#cbsa:35620` appeared. This is `New York-Northern New Jersey-Long Island, NY-NJ-PA` and likely didn't appear before because counties in it were entirely missing

https://github.com/covid-projections/covid-data-model/actions/runs/409039004 built snapshot 1409 from this branch.

Check for ICU headroom stats in the API output by running  `python ./pyseir/cli.py build-all --states NY --generate-api-v2 --output-dir output-tgb-pyseir-no-ny-agg` then looking at `jq .metrics.icuHeadroomRatio output-tgb-pyseir-no-ny-agg/county/36005.json`

Checked the place NYC is output by running `python ./pyseir/cli.py build-all --location-id-matches='.+\#fips\:(36061|3651000|36047|36081|36005|36085)' --generate-api-v2 --output-dir output-nyc` than looking at `output-nyc/place/3651000.json`.

### CODE REVIEW

**Review by ghop02 — COMMENTED** [review]
I think there are a couple more places that will need to be fixed:
[pyseir plotting name](https://github.com/covid-projections/covid-data-model/blob/master/pyseir/utils.py#L100-L101)

A couple places in `libs/pipeline.py` for completeness.
`libs/pipelines/api_v2_paths.py`
and you'll need to add another aggregate level to [here](https://github.com/covid-projections/covid-data-model/blob/master/libs/pipelines/api_v2_pipeline.py#L275-L277)

File: libs/datasets/timeseries.py [hunk 1]
```diff
@@ -1032,7 +1036,19 @@ def aggregate_regions(
...
+        # These are not equivalent so put the `value` attribute in static_agg.
+        static_agg[CommonFields.AGGREGATE_LEVEL] = (
+            static_agg.index.get_level_values(CommonFields.LOCATION_ID)
+            .map(pipeline.location_id_to_level)
+            .map(lambda l: l.value)
+        )
+    if CommonFields.FIPS in dataset_in.static.columns:
+        static_agg[CommonFields.FIPS] = static_agg.index.get_level_values(
+            CommonFields.LOCATION_ID
+        ).map(pipeline.location_id_to_fips)
```
Comment (ghop02) [inline libs/datasets/timeseries.py:1051]: 👍

**Review by ghop02 — APPROVED** [review]
Doesn't need to be in this PR, but we should also:
 - regenerate the schemas for the documentation (`.run.py api update-v2-schemas`).  
 - add places to the [api reference doc generation](https://github.com/covid-projections/covid-data-model/blob/master/api/update_open_api_spec.py).
 - Add an API update to [api updates](https://github.com/covid-projections/covid-data-model/blob/master/api/docs/docs/updates.md)

But this looks great! I do feel like we should wait to hear back from our partners who are relying on this before merging so that we can point them to the new places api.

**Comment by TomGoBravo** [comment]
> I think there are a couple more places that will need to be fixed:
> [pyseir plotting name](https://github.com/covid-projections/covid-data-model/blob/master/pyseir/utils.py#L100-L101)
> 
> A couple places in `libs/pipeline.py` for completeness.
> `libs/pipelines/api_v2_paths.py`
> and you'll need to add another aggregate level to [here](https://github.com/covid-projections/covid-data-model/blob/master/libs/pipelines/api_v2_pipeline.py#L275-L277)

I've fixed these. I think the only thing blocking doing the merge is getting word out consumers of the API.

**Pull request merged**